src/qpx_mmc.cpp

```diff
@@ -0,0 +1,4244 @@
+      //			drive->parms.speed_mult=drive->parms.max_read_speed_kb/drive->parms.max_read_speed_cd;
+      drive->parms.speed_mult = 176.4;
+      if (!drive->silent)
+        printf("Maximum CD speed: %dX, %5d kB/s; 1X = %.1f kB/s\nSpeeds:\n",
+               drive->parms.max_read_speed_cd, drive->parms.max_read_speed_kb,
+               drive->parms.speed_mult);
+      idx = 0;
+      prev_spd = 0;
+      for (spd = 1; ((idx < speed_tbl_size) &&
+                     (spd < (drive->parms.max_read_speed_cd + 2)));
+           spd++) {
+        //			for (spd=1; ((idx<speed_tbl_size) && (spd <
+        // 72)); spd++) {
+        drive->parms.read_speed_kb = (int)(spd * (drive->parms.speed_mult + 1));
+        if (!drive->silent)
+          printf("Trying:  %dX (%5d kB/s)\n", spd, drive->parms.read_speed_kb);
+        set_rw_speeds(drive);
+        get_rw_speeds(drive);
+        drive->parms.read_speed_cd =
+            (int)(drive->parms.read_speed_kb / drive->parms.speed_mult);
+        if (prev_spd != drive->parms.read_speed_cd) {
+          //					spd =
+          // drive->parms.read_speed_cd;
+          spd = max(spd, drive->parms.read_speed_cd);
+          drive->parms.speed_tbl[idx] = drive->parms.read_speed_cd;
+          drive->parms.speed_tbl_kb[idx] = drive->parms.read_speed_kb;
+          if (!drive->silent)
+            printf(" RD speed:  %dX (%5d kB/s)\n", drive->parms.speed_tbl[idx],
+                   drive->parms.speed_tbl_kb[idx]);
+          prev_spd = drive->parms.read_speed_cd;
+          idx++;
+          //					if (drive->capabilities &
+          // CAP_REAL_TIME_STREAMING) get_performance(drive);
+        }
+      }
+#ifdef speedidx
+    } else {
+      speedidx = -1;
+#endif
+    }
+  }
+  //	combo_Speed->clear();
+  idx = 0;
+#ifdef speedidx
+  if (!speedidx) {
+    while ((idx < speed_tbl_size) && (drive->parms.speed_tbl[idx] > 0)) {
+      int mlt = 176;
+      if (drive->media.type & DISC_DVD)
+        mlt = 1385;
+
+      //			combo_Speed->insertItem(QString().sprintf("%2dX
+      //(%dkB/s)",
+      //				drive->parms.speed_tbl[idx],mlt*drive->parms.speed_tbl[idx]));
+      if ((spd_kb / drive->parms.speed_mult) == drive->parms.speed_tbl[idx])
+        speedidx = idx;
+      idx++;
+    }
+  } else {
+    speedidx = 0;
+  }
+#endif
+  // 	combo_Speed->insertItem(tr("max"));
+
+  get_write_speed_tbl(drive);
+  for (int i = 0; i < speed_tbl_size && drive->parms.wr_speed_tbl_kb[i] > 0;
+       i++) {
+    if (!drive->silent)
+      printf(" WR speed:  %.1fX (%d kB/s)\n",
+             drive->parms.wr_speed_tbl_kb[i] / drive->parms.speed_mult,
+             drive->parms.wr_speed_tbl_kb[i]);
+  }
+
+  drive->parms.read_speed_kb = rspd_kb;
+  drive->parms.write_speed_kb = wspd_kb;
+  set_rw_speeds(drive);
+  //	combo_Speed->setCurrentItem(speedidx);
+  //	show_read_speed(drive);
+  return 0;
+}
+
+int get_write_speed_tbl(drive_info *drive) {
+  int offs;
+  int i, spdcnt;
```

Comment:
Codacy found an issue: [The scope of the variable 'i' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)